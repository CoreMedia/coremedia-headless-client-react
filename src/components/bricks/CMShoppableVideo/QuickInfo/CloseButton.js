// @flow
import React from 'react';
import PropTypes from 'prop-types';

import { IconButton as BasicIconButton } from '../../../basic/Button';
import CloseIcon from '../../../basic/Icon/CloseIcon';

const IconButton = BasicIconButton.extend`
  position: absolute;
  top: 5px;
  right: 5px;
`;

IconButton.displayName = 'IconButton';

type Props = {
  handleClick: (ev: SyntheticEvent<HTMLButtonElement>) => void,
};

const CloseButton = ({ handleClick }: Props) => (
  <IconButton onClick={handleClick} color="#808080">
    <CloseIcon />
  </IconButton>
);

CloseButton.propTypes = {
  handleClick: PropTypes.func.isRequired,
};

export default CloseButton;
