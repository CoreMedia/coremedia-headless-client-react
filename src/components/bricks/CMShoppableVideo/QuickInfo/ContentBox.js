// @flow
import { Box } from '../../../basic/Box';

const ContentBox = Box.extend`
  padding: 40px 40px 50px 40px;
`;

ContentBox.displayName = 'ContentBox';

export default ContentBox;
