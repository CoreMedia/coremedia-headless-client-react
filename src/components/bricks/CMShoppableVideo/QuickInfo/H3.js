// @flow
import { P } from '../../../basic/Paragraph';

const H3 = P.withComponent('h3').extend`
  margin: 20px 10px 10px 2px;
  align-self: start;
`;

H3.displayName = 'H3';

export default H3;
