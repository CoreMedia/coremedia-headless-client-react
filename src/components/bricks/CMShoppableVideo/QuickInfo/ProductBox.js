// @flow
import { Box } from '../../../basic/Box';
import { media } from '../../../../styles/themes/utils';

const ProductBox = Box.extend`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  ${props => media(props.theme.breakpoints.tablet)`
    flex-direction: row;
    align-items: start;
  `};
`;

ProductBox.displayName = 'ProductBox';

/*ProductBox.propTypes = {
  theme: PropTypes.shape({
    name: PropTypes.string.isRequired,
    breakpoints: PropTypes.shape({
      tablet: PropTypes.number.isRequired,
    }).isRequired,
  }),
};*/

ProductBox.defaultProps = {
  theme: {
    name: 'default',
    breakpoints: {
      tablet: 768,
    },
  },
};

export default ProductBox;
