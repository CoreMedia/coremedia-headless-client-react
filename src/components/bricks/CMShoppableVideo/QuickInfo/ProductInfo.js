// @flow
// import PropTypes from 'prop-types';

import { Box } from '../../../basic/Box';
import { media } from '../../../../styles/themes/utils';

const ProductInfo = Box.extend`
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 10px 0 0 0;
  ${props => media(props.theme.breakpoints.tablet)`
    align-items: start;
    padding: 10px 0 0 30px;
  `};
`;

ProductInfo.displayName = 'ProductInfo';

/*ProductInfo.propTypes = {
  theme: PropTypes.shape({
    name: PropTypes.string.isRequired,
    breakpoints: PropTypes.shape({
      tablet: PropTypes.number.isRequired,
    }).isRequired,
  }),
};*/

ProductInfo.defaultProps = {
  theme: {
    name: 'default',
    breakpoints: {
      tablet: 768,
    },
  },
};

export default ProductInfo;
