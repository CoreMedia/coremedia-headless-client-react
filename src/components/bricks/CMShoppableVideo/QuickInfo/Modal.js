// @flow
import PropTypes from 'prop-types';
import styled from 'styled-components';

import ReactModalAdapter from './ReactModalAdapter';
import { media } from '../../../../styles/themes/utils';

const Modal = styled(ReactModalAdapter).attrs({
  overlayClassName: {
    base: 'cm-overlay',
    afterOpen: 'cm-overlay--after-open',
    beforeClose: 'cm-overlay--before-close',
  },
  modalClassName: {
    base: 'cm-modal',
    afterOpen: 'cm-modal--after-open',
    beforeClose: 'cm-modal--before-close',
  },
  closeTimeoutMS: 200,
})`
  .cm-overlay {
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background-color: rgba(173, 173, 173, 0.9);
    cursor: pointer;
    z-index: 99999;
    opacity: 0;
    &--after-open {
      opacity: 1;
      transition: opacity 150ms ease-out;
    }
    &--before-close {
      opacity: 0;
      transition: opacity 150ms ease-in;
    }
  }
  .cm-modal {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background-color: #fff;
    box-sizing: content-box;
    cursor: default;
    z-index: 99999;
    transform: scale(0.5);
    &--after-open {
      transform: scale(1);
      transition: all 150ms ease-out;
    }
    &--before-close {
      transform: scale(0.5);
      transition: all 150ms ease-in;
    }
    ${props => media(props.theme.breakpoints.tablet)`
      top: 50%;
      left: 50%;
      width: 550px;
      height: auto;
      min-height: 365px;
      transform: translate(-50%, -50%) scale(0.5);
      box-shadow: 0px 0px 7px 0px rgba(0, 0, 0, 0.32);
      &--after-open {
        transform: translate(-50%, -50%) scale(1);
        transition: all 150ms ease-out;
      }
      &--before-close {
        transform: translate(-50%, -50%) scale(0.5);
        transition: all 150ms ease-in;
      }
    `};
  }
`;

Modal.displayName = 'Modal';

Modal.propTypes = {
  theme: PropTypes.shape({
    breakpoints: PropTypes.shape({
      tablet: PropTypes.number.isRequired,
    }).isRequired,
  }),
};

Modal.defaultProps = {
  theme: {
    breakpoints: {
      tablet: 768,
    },
  },
};

export default Modal;
