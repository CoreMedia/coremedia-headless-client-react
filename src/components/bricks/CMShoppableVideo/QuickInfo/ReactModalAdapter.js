// @flow
import React from 'react';
import PropTypes from 'prop-types';
import Modal from 'react-modal';

type Props = {
  className: string,
  overlayClassName: string,
  modalClassName: string,
};

const ReactModalAdapter = ({ className, overlayClassName, modalClassName, ...props }: Props) => (
  <Modal
    overlayClassName={overlayClassName}
    className={modalClassName}
    portalClassName={className}
    ariaHideApp={false}
    {...props}
  />
);

ReactModalAdapter.propTypes = {
  className: PropTypes.string.isRequired,
  overlayClassName: PropTypes.shape({
    base: PropTypes.string.isRequired,
    afterOpen: PropTypes.string.isRequired,
    beforeClose: PropTypes.string.isRequired,
  }).isRequired,
  modalClassName: PropTypes.shape({
    base: PropTypes.string.isRequired,
    afterOpen: PropTypes.string.isRequired,
    beforeClose: PropTypes.string.isRequired,
  }).isRequired,
};

export default ReactModalAdapter;
