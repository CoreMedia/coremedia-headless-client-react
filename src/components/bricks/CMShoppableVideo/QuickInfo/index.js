// @flow
import React from 'react';
import PropTypes from 'prop-types';
//import { injectGlobal } from 'styled-components';

import Modal from './Modal';
import Picture from '../../CMPicture';
import { Button } from '../../../basic/Button';
import { P } from '../../../basic/Paragraph';
import CloseButton from './CloseButton';
import Title from './Title';
import ContentBox from './ContentBox';
import ProductInfo from './ProductInfo';
import ProductBox from './ProductBox';
import H3 from './H3';

/*injectGlobal`
  .ReactModal__Body--open {
    overflow: hidden;
  }
`;*/

type Props = {
  item: ?Object,
  handleClose: (ev: SyntheticEvent<HTMLButtonElement>) => void,
};

const QuickInfo = ({ item, handleClose }: Props) => (
  <Modal isOpen={!!item} contentLabel="Minimal Modal Example" onRequestClose={handleClose}>
    <CloseButton handleClick={handleClose} />
    {item && (
      <ContentBox>
        <Title>{item.teaserTitle}</Title>
        <ProductBox>
          <Picture
            link={item.pictureLink}
            ratio="quickview_ratio5x6"
            title={item.pictureTitle}
            alt={item.pictureAlt}
          />
          <ProductInfo>
            <P>{item.teaserText}</P>
            <H3>{`${item.teaserTitle} EUR ${item.price}`}</H3>
            <Button>order</Button>
          </ProductInfo>
        </ProductBox>
      </ContentBox>
    )}
  </Modal>
);

QuickInfo.propTypes = {
  item: PropTypes.object,
  handleClose: PropTypes.func.isRequired,
};

export default QuickInfo;
