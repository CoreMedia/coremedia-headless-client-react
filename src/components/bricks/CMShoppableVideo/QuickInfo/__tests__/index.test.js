// @flow
import React from 'react';
import 'jest-styled-components';

import QuickInfo from '..';

describe('QuickInfo Component', () => {
  it('should render correctly', () => {
    const item = {
      teaserTitle: 'Title',
      teaserText: 'Text',
      pictureLink: 'Link',
      pictureTitle: 'Picture Title',
      pictureAlt: 'Picture Alt',
      price: '100',
    };
    const handleClose = jest.fn().mockName('handleClose');
    const wrapper = shallow(<QuickInfo item={item} handleClose={handleClose} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should execute handleClose', () => {
    const item = {
      teaserTitle: 'Title',
      teaserText: 'Text',
      pictureLink: 'Link',
      pictureTitle: 'Picture Title',
      pictureAlt: 'Picture Alt',
      price: '100',
    };
    const handleClose = jest.fn().mockName('handleClose');
    const wrapper = shallow(<QuickInfo item={item} handleClose={handleClose} />);
    wrapper.simulate('requestClose');
    expect(handleClose).toHaveBeenCalled();
  });
});
