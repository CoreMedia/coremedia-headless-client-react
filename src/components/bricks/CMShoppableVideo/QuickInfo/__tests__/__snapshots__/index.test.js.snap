// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuickInfo Component should render correctly 1`] = `
<Modal
  contentLabel="Minimal Modal Example"
  isOpen={true}
  onRequestClose={[MockFunction handleClose]}
  theme={
    Object {
      "breakpoints": Object {
        "tablet": 768,
      },
    }
  }
>
  <CloseButton
    handleClick={[MockFunction handleClose]}
  />
  <ContentBox
    theme={
      Object {
        "name": "default",
      }
    }
  >
    <Title
      theme={
        Object {
          "breakpoints": Object {
            "desktop": 992,
            "large": 1280,
            "tablet": 768,
          },
          "name": "default",
        }
      }
    >
      Title
    </Title>
    <ProductBox
      theme={
        Object {
          "breakpoints": Object {
            "tablet": 768,
          },
          "name": "default",
        }
      }
    >
      <WithTheme(Picture)
        alt="Picture Alt"
        link="Link"
        ratio="quickview_ratio5x6"
        theme={
          Object {
            "breakpoints": Object {
              "desktop": 992,
              "large": 1280,
              "small": 480,
              "tablet": 768,
            },
          }
        }
        title="Picture Title"
      />
      <ProductInfo
        theme={
          Object {
            "breakpoints": Object {
              "tablet": 768,
            },
            "name": "default",
          }
        }
      >
        <P
          theme={
            Object {
              "breakpoints": Object {
                "desktop": 992,
                "large": 1280,
                "tablet": 768,
              },
              "name": "default",
            }
          }
        >
          Text
        </P>
        <H3
          theme={
            Object {
              "breakpoints": Object {
                "desktop": 992,
                "large": 1280,
                "tablet": 768,
              },
              "name": "default",
            }
          }
        >
          Title EUR 100
        </H3>
        <Button
          theme={
            Object {
              "breakpoints": Object {
                "desktop": 992,
                "large": 1280,
                "small": 480,
                "tablet": 768,
              },
              "name": "default",
            }
          }
        >
          order
        </Button>
      </ProductInfo>
    </ProductBox>
  </ContentBox>
</Modal>
`;
